tensorboardX/record_writer.py

```diff
@@ -75,12 +76,11 @@ def write(self, val):
         self.buffer.write(val)
 
     def flush(self):
-        s3 = boto3.client('s3')
-        bucket, path = self.bucket_and_path()
-        s3.upload_fileobj(self.buffer, bucket, path)
```

Comment:
As a quick fix you can just do

```
s3.upload_fileobj(self.buffer, bucket, path)
self.buffer = io.BytesIO()
```

Reply:
I think I tried that but it woudn't append... I can't remember now but I tried a few things around that vein and then started looking into "streaming" data to S3 and found that `smart_open` package

Reply:
I will try a bit later, today I am swamped with meetings.

Reply:
@andrewkho 

got this working with the following implementation:
```
    def flush(self):
        s3 = boto3.client('s3')
        bucket, path = self.bucket_and_path()
        upload_buffer = copy.copy(self.buffer)
        upload_buffer.seek(0)
        s3.upload_fileobj(upload_buffer, bucket, path)
```

the trick was to copy `self.buffer` so that it would remain open, while the `upload_buffer` is uploaded to s3 and closed.